Core/include/Acts/TrackFitting/Chi2Fitter.hpp

```diff
@@ -632,6 +644,18 @@ class Chi2Fitter {
     // the result object which will be returned. Overridden every iteration.
     Chi2Result c2r;
 
+    if (not trajectory) {
+      trajectory = std::make_shared<traj_t>();
+    };
+
+    using paramType = typename std::conditional<
```

Comment:
I think we can remove this check as well, since we are fixed to `BoundTrackParameters` for now

Reply:
(does not apply if you go back to templated case)

Reply:
Hmm this conditional acutally makes no sense that way, `std::is_same<BoundTrackParameters, BoundTrackParameters>::value` is always true...